apps/gateway/src/middlewares/providers/bedrock.ts

```diff
@@ -33,6 +34,50 @@ export class BedrockProviderAdapter
     "anthropic/claude-opus-4-5-v1": "anthropic.claude-opus-4-5-20251101-v1:0",
   };
 
+  private static toSnakeCase(str: string): string {
+    return str.replaceAll(/[A-Z]/g, (letter) => `_${letter.toLowerCase()}`);
+  }
+
+  private static convertObjectKeysToSnakeCase(
+    obj: Record<string, any>,
+  ): Record<string, any> {
+    const newObj: Record<string, any> = {};
+    for (const key in obj) {
+      if (Object.prototype.hasOwnProperty.call(obj, key)) {
+        const originalValue = obj[key];
+        const convertedValue =
+          typeof originalValue === "object" &&
+          originalValue !== null &&
+          !Array.isArray(originalValue)
+            ? BedrockProviderAdapter.convertObjectKeysToSnakeCase(originalValue)
+            : originalValue;
+        newObj[BedrockProviderAdapter.toSnakeCase(key)] = convertedValue;
+      }
+    }
+    return newObj;
+  }
+
+  transformOptions(options?: ProviderOptions): ProviderOptions {
+    const { modelConfig, ...rest } = options || {};
+
+    let modifiedOptions: ProviderOptions = { ...rest };
+
+    if (modelConfig) {
+      const snakeCaseConfig =
+        BedrockProviderAdapter.convertObjectKeysToSnakeCase(
+          modelConfig as Record<string, any>,
+        );
+      modifiedOptions = {
+        ...modifiedOptions,
+        bedrock: {
```

Comment:
Related to my comments on the transformOptions hook in models, we might need to infer the term "bedrock" from the external caller. This could be done from the slug, or, since sometimes the slug and the key for the provider options might vary, introduce an additional field for the providerOptions key that can be used from the aiModelFactory.

Reply:
Make sense, let me do that.